python/python/lance/dataset.py

```diff
@@ -1598,6 +1608,60 @@ def optimize_indices(self, **kwargs):
         self._dataset._ds.optimize_indices(**kwargs)
 
 
+class LanceStats:
+    def __init__(self, dataset: _Dataset):
+        self._ds = dataset
+
+    def dataset_stats(self, max_rows_per_group: int = 1024) -> Dict[str, Any]:
```

Comment:
Would be nice if the return type was typed so auto-completion. There is a typeddict option starting in Python 3.8: https://peps.python.org/pep-0589/

```suggestion
    def dataset_stats(self, max_rows_per_group: int = 1024) -> Dict[str, Any]:
```

Reply:
Thanks good suggestion! made this change